### PR TITLE
revamp sitrep and nflverse_data class

### PR DESCRIPTION
**Pull request by mrcaseb**
This is a duplicate of #198 

I moved the code into a fresh branch because git couldn't identify actual code changes (maybe encoding or line ending)

### CODE REVIEW

File: R/utils_sitrep.R [hunk 1]
```diff
@@ -49,52 +49,141 @@ ffverse_sitrep <- function(pkg = c("ffscrapr","ffsimulator","ffpros","ffopportun
...
+  if (length(x$dependencies) >= 1) {
+    cli::cat_rule(cli::style_bold("Package Dependencies"), col = cli::make_ansi_style("cyan"), line = 1)
+    .cat_packages(x$dependencies$package, x$dependencies$version)
+  }
+  if (length(x$not_installed) >= 1) {
+    cli::cat_rule(cli::style_bold("Not Installed"), col = cli::make_ansi_style("cyan"), line = 1)
+    .cat_packages(x$not_installed, rep_len("", length(x$not_installed)))
+  }
 
-  if(redact_path) package_opts[grepl("path", names(package_opts))] <- "{redacted, use redact_path = FALSE to show}"
```
Comment (tanho63) [inline R/utils_sitrep.R:82]: oops, I think I lost the redact path portion here. how much did we end up needing it?

Comment (tanho63) [reply R/utils_sitrep.R:82]: @mrcaseb

Comment (mrcaseb) [reply R/utils_sitrep.R:82]: I'd prefer keeing it unless it is a lot of work to implement now

File: R/utils_sitrep.R [hunk 2]
```diff
@@ -49,52 +49,141 @@ ffverse_sitrep <- function(pkg = c("ffscrapr","ffsimulator","ffpros","ffopportun
...
+  out$installed <- lapply(s$otherPkgs, `[`, c("Package", "Version"))
+  out$installed <- as.data.frame(data.table::rbindlist(out$installed))
+  out$installed <- stats::setNames(out$installed, c("package","version"))
 
   # Relies on options for a given package being prefixed by said package name
   opts <- options()
-  package_opts <- opts[Reduce(lapply(packages, function(package) grepl(package, x = names(opts), fixed = TRUE)),f = `|`)]
+  pkg_search_string <- paste(packages, collapse = "|")
+  out$package_options <- opts[grepl(pkg_search_string, x = names(opts))]
+
```
Comment (tanho63) [inline R/utils_sitrep.R:92]:
```suggestion
  out$package_options <- opts[grepl(pkg_search_string, x = names(opts))]
if(redact_path) out$package_options[grepl("path|token", names(out$package_options))] <- "{redacted, use redact_path = FALSE to show}"
```

**Review by tanho63 — DISMISSED** [review]

**Review by tanho63 — APPROVED** [review]

**Comment by tanho63** [comment]
may need to bypass merge checks and then update which checks are being keyed on again

File: .github/workflows/R-CMD-check.yaml [hunk 3]
```diff
@@ -1,70 +1,51 @@
...
 
     name: ${{ matrix.config.os }} (${{ matrix.config.r }})
 
     strategy:
       fail-fast: false
       matrix:
         config:
-          - {os: windows-latest, r: 'release', rspm: "https://packagemanager.rstudio.com/all/latest"}
-          #- {os: windows-latest, r: '3.6',     rspm: "https://packagemanager.rstudio.com/cran/latest"}
-          - {os: macOS-latest,   r: 'release', rspm: "https://packagemanager.rstudio.com/all/latest"}
```
[the remark this answers is not in the document]
Comment (mrcaseb) [reply .github/workflows/R-CMD-check.yaml:30]: yeah that's what I figured and why I decided to test a newer workflow. I hope it resolves our problems and stops annoying us

**Comment by mrcaseb** [comment]
changed required checks in branch protection rules. We could merge now but we found some problems with pak

**Review by tanho63 — APPROVED** [review]

**Pull request merged**